### PR TITLE
[codex] Resolve UI issues

### LINKED ISSUE #17: Support null field filters in Firestore query builder
## Problem
It is not obvious how to filter documents where a field is `null`.

## Scope
Make null filtering discoverable and usable in the Firestore query UI.

## Acceptance criteria
- Users can build a query for `field == null` from the UI.
- Validation handles null values explicitly.
- UI copy makes the behavior clear if Firestore has limitations for other null-related operators.
- Add tests for building and running a null filter query.

### LINKED ISSUE #18: Add collection metadata management UI
## Problem
Collection metadata can remain stored indefinitely, with no clear way to inspect or delete it.

## Scope
Add a UI for visualizing stored collection metadata and letting users remove it when they choose.

## Acceptance criteria
- Users can view stored collection metadata by project, connection, and path.
- Users can delete selected metadata entries.
- Deleting metadata does not delete Firestore data.
- Empty, error, and loading states are explicit.
- Add tests for listing and deleting metadata.

### LINKED ISSUE #19: Review sticky behavior for Settings page header
## Problem
The Settings page header scrolls away with the page. It may be harder to keep context and access top-level actions.

## Scope
Decide whether the Settings header should be sticky and implement the chosen behavior.

## Acceptance criteria
- Settings header behavior is intentional and consistent with app layout patterns.
- Header actions remain accessible when useful.
- No content overlaps or layout jumps while scrolling.
- Add or adjust UI tests if behavior changes.

### LINKED ISSUE #20: Make density setting visibly affect the app
## Problem
Changing the density setting does not produce a noticeable difference.

## Scope
Audit where density is applied and make the setting visibly affect relevant UI surfaces.

## Acceptance criteria
- Density impact is visible on core surfaces like tables, lists, drawers, panels, and forms.
- Density tokens are applied consistently through the design system where practical.
- Settings preview or copy makes the behavior understandable.
- Add tests for density class or state application where useful.

### LINKED ISSUE #21: Prevent status bar items from wrapping on narrow windows
## Problem
Status bar items can render over multiple lines when the window is narrow.

## Scope
Make status bar items single-line and truncate text when needed.

## Acceptance criteria
- Status bar remains one line at narrow widths.
- Long labels truncate cleanly with accessible full text via title or tooltip where useful.
- Items do not overlap.
- Add responsive UI coverage for narrow widths.

### LINKED ISSUE #23: Polish tab context menu styling and actions
## Problem
The tab right-click menu looks visually uneven. Padding and icon usage are inconsistent, and `Switch tab` may be unnecessary.

## Scope
Review tab context menu content and styling.

## Acceptance criteria
- Menu padding and spacing match the design system.
- Icon usage is consistent: either all actionable items have icons or none do.
- `Switch tab` is removed if redundant.
- Keyboard and pointer interactions still work.
- Add or update component tests for the final actions.

### LINKED ISSUE #24: Remove duplicate production labels and only flag emulator targets
## Problem
Production labels appear in multiple visible places, including tabs and status bar. Production should be the default, and only emulator/local targets should be flagged.

## Scope
Simplify environment labeling across tabs, headers, and status bar.

## Acceptance criteria
- Production is not shown as a repeated badge or label.
- Emulator/local targets are clearly flagged.
- The same environment label is not shown in multiple visible places at once.
- Update tests for target mode rendering.

### PR DESCRIPTION
**Pull request by viniciusrmcarneiro**
## Summary
- Add null filter CTA in Firestore query builder.
- Make Settings header sticky, density visible, status bar single-line, and tab menu consistent.
- Show emulator badges only, no production badges.
- Add Settings UI to view and clear saved Firestore metadata.

## Skipped
- #22 and #25 need release/update channel decisions.

## Checks
- pnpm format:check
- pnpm lint
- pnpm typecheck
- pnpm test
- pnpm build

Fixes #17
Fixes #18
Fixes #19
Fixes #20
Fixes #21
Fixes #23
Fixes #24

### CODE REVIEW

File: packages/ui/src/explorer-tree/ExplorerTree.tsx [hunk 1]
```diff
@@ -37,7 +39,8 @@ export function ExplorerTree<TNode extends ExplorerTreeRowModel = ExplorerTreeRo
   {
     className,
     contextMenu,
-    estimateSize = () => 32,
+    density,
+    estimateSize,
     onOpen,
     onSelect,
```
[the remark this answers is not in the document]
Comment (viniciusrmcarneiro) [reply packages/ui/src/explorer-tree/ExplorerTree.tsx:45]: Fixed by defaulting ExplorerTree virtualization to the density tree row height and using that same value for row min-height. Added regression coverage.

File: packages/product-ui/src/settings-dialog/SettingsDialog.tsx [hunk 2]
```diff
@@ -407,6 +529,33 @@ function SettingsSummary({ snapshot }: { readonly snapshot: SettingsSnapshot | n
...
+  const keys = new Set([
+    ...Object.keys(snapshot.firestoreFieldCatalogs),
+    ...Object.keys(snapshot.resultTableLayouts),
+  ]);
+  const sortedKeys: string[] = [];
+  for (const key of keys) {
+    const nextIndex = sortedKeys.findIndex((existing) => key.localeCompare(existing) < 0);
+    if (nextIndex === -1) sortedKeys.push(key);
+    else sortedKeys.splice(nextIndex, 0, key);
+  }
```
[the remark this answers is not in the document]
Comment (viniciusrmcarneiro) [reply packages/product-ui/src/settings-dialog/SettingsDialog.tsx]: Leaving this as-is. The repo lint forbids Array#sort through unicorn/no-array-sort, and the current TS target does not support toSorted. The current insertion keeps ordering deterministic without adding a broader helper.

[the remark this answers is not in the document]
Comment (viniciusrmcarneiro) [reply packages/product-ui/src/settings-dialog/SettingsDialog.tsx]: Fixed with a lint-compliant merge sort over the metadata keys and added ordering coverage.

File: packages/ui/src/input/Input.tsx [hunk 3]
```diff
@@ -7,7 +7,7 @@ const inputVariants = cva(
   {
     variants: {
       density: {
-        compact: 'h-[var(--density-compact-control-height)] text-sm',
+        compact: 'h-[var(--density-control-height)] text-sm',
         comfortable: 'h-[var(--density-comfortable-control-height)] text-sm',
       },
```
[the remark this answers is not in the document]
Comment (viniciusrmcarneiro) [reply packages/ui/src/input/Input.tsx:12]: Fixed by restoring compact to the concrete compact token while keeping comfortable on the concrete comfortable token. Added variant coverage.

File: apps/desktop/src/renderer/app/hooks/useAppShellController.ts [hunk 4]
```diff
@@ -110,6 +112,15 @@ export function useAppShellController(
     selectedTreeItemId: selection.treeItemId,
     setLastAction,
   });
+  useEffect(() => {
+    let cancelled = false;
+    void repositories.settings.load().then((snapshot) => {
+      if (!cancelled) setDensity(snapshot.density);
+    });
```
[the remark this answers is not in the document]
Comment (viniciusrmcarneiro) [reply apps/desktop/src/renderer/app/hooks/useAppShellController.ts]: Fixed by catching the settings load rejection, falling back to default density, and surfacing the failure in status. Added hook coverage.
